crates/extrinsics/src/contract_storage.rs

```diff
@@ -0,0 +1,312 @@
+// Copyright 2018-2023 Parity Technologies (UK) Ltd.
+// This file is part of cargo-contract.
+//
+// cargo-contract is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// cargo-contract is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with cargo-contract.  If not, see <http://www.gnu.org/licenses/>.
+
+use anyhow::Result;
+use ink_metadata::{
+    layout::Layout,
+    InkProject,
+};
+use scale_info::form::PortableForm;
+use serde::Serialize;
+use sp_core::storage::ChildInfo;
+use std::{
+    collections::BTreeMap,
+    fmt::Display,
+};
+use subxt::{
+    backend::{
+        legacy::{
+            rpc_methods::Bytes,
+            LegacyRpcMethods,
+        },
+        rpc::{
+            rpc_params,
+            RpcClient,
+        },
+        BlockRef,
+    },
+    error::DecodeError,
+    ext::scale_decode::{
+        IntoVisitor,
+        Visitor,
+    },
+    Config,
+    OnlineClient,
+};
+
+use super::{
+    fetch_contract_info,
+    url_to_string,
+    Client,
+    ContractInfo,
+    DefaultConfig,
+    TrieId,
+};
+
+pub struct ContractStorage<C: Config = DefaultConfig> {
+    rpc: ContractStorageRpc<C>,
+}
+
+impl<C: Config> ContractStorage<C>
+where
+    C::AccountId: AsRef<[u8]> + Display + IntoVisitor,
+    DecodeError: From<<<C::AccountId as IntoVisitor>::Visitor as Visitor>::Error>,
+    BlockRef<sp_core::H256>: From<C::Hash>,
```

Comment:
Maybe we should just return BlockRef<sp_core::H256> from get_best_block until we figure out how to make it cleaner.

Reply:
I don't see the problem with the current approach using generics

Reply:
I've removed these bounds now, it was because I was still using the `pub type Client = OnlineClient<DefaultConfig>` alias. I have switched that to using the generic `OnlineClient` which has fixed it.